source/common/router/router_ratelimit.cc

```diff
@@ -124,6 +124,30 @@ bool DynamicMetaDataAction::populateDescriptor(
   return false;
 }
 
+RouteEntryMetaDataAction::RouteEntryMetaDataAction(
+    const envoy::config::route::v3::RateLimit::Action::RouteEntryMetaData& action)
+    : metadata_key_(action.metadata_key()), descriptor_key_(action.descriptor_key()),
+      default_value_(action.default_value()) {}
+
+bool RouteEntryMetaDataAction::populateDescriptor(const Router::RouteEntry& route,
+                                                  RateLimit::Descriptor& descriptor,
+                                                  const std::string&, const Http::HeaderMap&,
+                                                  const Network::Address::Instance&,
+                                                  const envoy::config::core::v3::Metadata*) const {
+  const ProtobufWkt::Value& metadata_value =
+      Envoy::Config::Metadata::metadataValue(&route.metadata(), metadata_key_);
+
+  if (!metadata_value.string_value().empty()) {
+    descriptor.entries_.push_back({descriptor_key_, metadata_value.string_value()});
+    return true;
+  } else if (metadata_value.string_value().empty() && !default_value_.empty()) {
```

Comment:
```suggestion
  } else if (!default_value_.empty()) {
```

technically not needed, but if you want to keep it for expressiveness that's fine

Reply:
I would keep it as it is, because it is telling the whole story better.